src/waveform/waveformwidgetfactory.cpp

```diff
@@ -223,6 +224,13 @@ bool WaveformWidgetFactory::setConfig(UserSettingsPointer config) {
         m_config->set(ConfigKey("[Waveform]","ZoomSynchronization"), ConfigValue(m_zoomSync));
     }
 
+    int displayGrid = m_config->getValueString(ConfigKey("[Waveform]", "beatGridLinesCheckBox")).toInt(&ok);
```

Comment:
Later we may use other controls, but the config key remains. 
how about just "showBeatgrid"? 

By the way, you can use here the m_config->getValue() template function.